### PR TITLE
Utility components

### PR DESCRIPTION
**Pull request by osdiab**
A general purpose, flexible and themed button; and an easier to use internationalization component.

This is a necessary prerequisite for #103

### CODE REVIEW

File: src/components/IntlMessage.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+type Props =
+  | BaseProps
+  | (BaseProps & (RenderByTagNameProps | CustomRendererProps));
+
+/**
+ * Neat component for rendering internationalized text in a variety of ways.
+ *
+ * react-intl's FormattedMessage class does everything we need, but the syntax
+ * for rendering in various ways is not always straightforward.
```
Comment (rahulgi) [inline src/components/IntlMessage.tsx]: Could you add an example of how to use this component to this comment?

File: src/components/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+      props.theme.type === ButtonType.LINK
+        ? buttonStyles.hover.textDecoration[props.theme.type]
+        : "none"};
+  }
+`;
+
+/**
+ * General purpose button that matches application styles.
+ * Not intended for webpage linking, but instead for running
+ * code on click.
```
Comment (rahulgi) [inline src/components/Button.tsx]: Could you add an example of how to use this component to this comment?

Comment (osdiab) [reply src/components/Button.tsx]: 8c2ec18

File: src/components/IntlMessage.tsx (hunk 1 above)
Comment (osdiab) [reply src/components/IntlMessage.tsx]: 8c2ec18

**Pull request merged**